RecoBTag/DeepFlavour/plugins/DeepFlavourExporter.cc

```diff
@@ -0,0 +1,133 @@
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
+
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+
+#include "DataFormats/PatCandidates/interface/Jet.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/Framework/interface/makeRefToBaseProdFrom.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "DataFormats/BTauReco/interface/JetTag.h"
+#include "DataFormats/DeepFormats/interface/DeepFlavourTagInfo.h"
+
+#include <algorithm>
+#include <string>
+
+#include "TTree.h"
+
+class DeepFlavourExporter : public edm::one::EDAnalyzer<edm::one::SharedResources> {
+
+  public:
+	  explicit DeepFlavourExporter(const edm::ParameterSet&);
+	  ~DeepFlavourExporter();
+
+	  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+  private:
+
+    typedef std::vector<reco::DeepFlavourTagInfo> TagInfoCollection;
+    typedef reco::JetTagCollection JetTagCollection;
+
+    virtual void beginJob() override;
+    virtual void analyze(const edm::Event&, const edm::EventSetup&) override;
+    virtual void endJob() override;
+
+	  const edm::EDGetTokenT< TagInfoCollection > tag_info_src_;
+	  edm::EDGetTokenT<edm::View<pat::Jet>> jet_src_;
+
+    // to keep data to be saved in the TTree
+    std::vector<std::string> disc_names_;
+    std::vector<float> disc_values_;
+    deep::DeepFlavourFeatures features_;
+    
+    bool use_jet_src_;
+
+    edm::Service<TFileService> fs;
+    TTree* tree;
+};
+
+DeepFlavourExporter::DeepFlavourExporter(const edm::ParameterSet& iConfig) :
+  tag_info_src_(consumes<TagInfoCollection>(iConfig.getParameter<edm::InputTag>("tag_info_src"))),
+  disc_names_((iConfig.getParameter<std::vector<std::string>>("btagDiscriminators"))),
+  disc_values_(disc_names_.size(),0.0)
+{
+  auto jet_src_tag = iConfig.getParameter<edm::InputTag>("jet_src");
+  if (jet_src_tag.label().size() == 0) {
+    use_jet_src_ = false;
+  } else {
+    jet_src_ = consumes<edm::View<pat::Jet>>(jet_src_tag);
+  } 
+  usesResource("TFileService");
+}
+
+
+DeepFlavourExporter::~DeepFlavourExporter()
+{
+}
+
+void DeepFlavourExporter::fillDescriptions(edm::ConfigurationDescriptions& descriptions)
+{
+    edm::ParameterSetDescription desc;
+    desc.setUnknown();
+    descriptions.addDefault(desc);
```

Comment:
better fill this with something meaningful and then perhaps drop DeepFlavourExporter_cfi.py file which will be made by the framework instead

Reply:
I will remove the Exporter plugin becuase it is not needed anymore

Reply:
Exporter removed with https://github.com/cms-sw/cmssw/pull/19893/commits/de7e130c318f4dc84a506c2e2aef7d32900013a8